contracts/token/erc1155/abstract/ImmutableERC1155Base.sol

```diff
@@ -0,0 +1,255 @@
+        }
+        return admins;
+    }
+
+    /** 
+     * @notice Override of setApprovalForAll from {ERC721}, with added Allowlist approval validation
+     * @param operator The address to approve as an operator for the caller.
+     * @param approved True if the operator is approved, false to revoke approval.
+     */
+    function setApprovalForAll(address operator, bool approved) public override(ERC1155) validateApproval(operator) {
+        super.setApprovalForAll(operator, approved);
+    }
+
+    /**
+     * @notice Override of _safeTransferFrom from {ERC1155}, with added Allowlist transfer validation
+     * @param from The current owner of the token.
+     * @param to The new owner.
+     * @param id The token identifier to transfer.
+     * @param value The amount to transfer.
+     * @param data Additional data with no specified format, sent in call to `to`.
+     */
+    function _safeTransferFrom(address from, address to, uint256 id, uint256 value, bytes memory data) internal override validateTransfer(from, to) {
+        super._safeTransferFrom(from, to, id, value, data);
+    }
+
+    /**
+     * @notice Override of _safeBatchTransferFrom from {ERC1155}, with added Allowlist transfer validation
+     * @param from The current owner of the token.
+     * @param to The new owner.
+     * @param ids The token identifiers to transfer.
+     * @param values The amounts to transfer per token id.
+     * @param data Additional data with no specified format, sent in call to `to`.
+     */
+    function _safeBatchTransferFrom(address from, address to, uint256[] memory ids, uint256[] memory values, bytes memory data) internal override validateTransfer(from, to) {
+        super._safeBatchTransferFrom(from, to, ids, values, data);
+    }
+
+    /**
+     * @notice sets the default royalty receiver
+     * @param receiver The address of the royalty receiver
+     * @param feeNumerator The royalty fee numerator
+     */
+    function setDefaultRoyaltyReceiver(address receiver, uint96 feeNumerator) public onlyRole(DEFAULT_ADMIN_ROLE) {
+        _setDefaultRoyalty(receiver, feeNumerator);
+    }
+
+    /** 
+     * @notice Set the royalty receiver address for a specific tokenId 
+     * @param tokenId The token identifier to set the royalty receiver for.
+     * @param receiver The address of the royalty receiver
+     * @param feeNumerator The royalty fee numerator
+    */
+    function setNFTRoyaltyReceiver(
+        uint256 tokenId,
+        address receiver,
+        uint96 feeNumerator
+    ) public onlyRole(MINTER_ROLE) {
+        _setTokenRoyalty(tokenId, receiver, feeNumerator);
+    }
+
+    /** 
+     * @notice Set the royalty receiver address for a list of tokenIDs 
+     * @param tokenIds The token identifiers to set the royalty receiver for.
+     * @param receiver The address of the royalty receiver
+     * @param feeNumerator The royalty fee numerator
+    */
+    function setNFTRoyaltyReceiverBatch(
```

Comment:
Just flagging that we should remember to rename this once we decide what do call 1155s